docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -164,27 +179,33 @@ Accordingly, a top-down approach to ``apply`` might look like this:
 .. code-block:: python
 
     def apply(self, transaction, context):
-        signer, game_name, action, space = \
-            self._unpack_transaction(transaction)
 
-        board, state, player1, player2 = \
-            self._get_state_data(game_name, context)
+        header = transaction.header
+        signer = header.signer_public_key
+
+        xo_payload = XoPayload.from_bytes(transaction.payload)
 
-        updated_game_data = self._play_xo(
-            board, state,
-            player1, player2,
-            signer, action, space
-        )
+        xo_state = XoState(context)
 
-        self._store_game_data(game_name, updated_game_data, context)
+        if xo_payload.action == 'delete':
+            ...
+        elif xo.payload.action == 'create':
+            ...
+        elif xo.payload.action == 'take':
+            ...
+        else:
+            raise InvalidTransaction('Unhandled action: {}'.format(
+                xo_payload.action))
 
 {% endif %}
 
-Note that the third step is the only one that actually concerns tic-tac-toe;
-the other three steps all concern the coordination of data.
+For every new payload the transaction processor validates rules surrounding deleting
+a game, creating a game, and taking a new space. If all of the rules validate, then
+state is updated based on whether we are deleting or creating a game or updating the
```

Comment:
Please change "deleting or creating a game or" to:

     creating a game, deleting a game, or 

(This matches the action order used above and below, and adds some needed punctuation.)